src/genie/libs/parser/iosxe/show_cable_tdr_interface.py

```diff
@@ -0,0 +1,164 @@
+'''show_cable_tdr_interface.py
+IOSXE parser for the following show command
+	* show cable tdr interface {interface}
+'''
+
+# Python
+import re
+# import unittest
+
+# Metaparser
+from genie.metaparser import MetaParser
+from genie.metaparser.util.schemaengine import Schema, Any, Or, ListOf, \
+        Optional, And, Default, Use
+
+# import parser utils
+from genie.libs.parser.utils.common import Common
+
+# =================================================
+# Schema for:
+#   * 'show cable tdr interface {interface}'
+# ==================================================
+class ShowCableTdrIntSchema(MetaParser):
+    
+    schema = {
+        Any(): {
+            Optional('interface'): str,
+            Optional('speed'): str,
+            Optional('date'): str,
+            Optional('time'): str,
+            Optional('pair_a'): {
+                Optional('length'): int,
+                Optional('tolerance'): int,
+                Optional('remote_pair'): str,
+                Optional('status'): str,
+            },
+            Optional('pair_b'): {
+                Optional('length'): int,
+                Optional('tolerance'): int,
+                Optional('remote_pair'): str,
+                Optional('status'): str,
+            },
+            Optional('pair_c'): {
+                Optional('length'): int,
+                Optional('tolerance'): int,
+                Optional('remote_pair'): str,
+                Optional('status'): str,
+            },
+            Optional('pair_d'): {
+                Optional('length'): int,
+                Optional('tolerance'): int,
+                Optional('remote_pair'): str,
+                Optional('status'): str,
+            },
```

Comment:
Instead of having 4 keys with repeated values, why not have a list of pairs? In this case, the key to each pair would be its name
```suggestion
            'pairs': {
                Any() : {
                    Optional('length'): int,
                    Optional('tolerance'): int,
                    Optional('remote_pair'): str,
                    Optional('status'): str,
               }
           }            
```